src/Docfx.Build/TableOfContents/BuildTocDocument.cs

```diff
@@ -24,6 +23,32 @@ class BuildTocDocument : BaseDocumentBuildStep
     /// </summary>
     public override IEnumerable<FileModel> Prebuild(ImmutableList<FileModel> models, IHostService host)
     {
+
+        if (!models.Any())
+        {
+            return TocHelper.ResolveToc(models.ToImmutableList());
+        }
+
+        // Keep auto toc agnostic to the toc file naming convention.
+        var tocFileName = models.First().Key.Split('/').Last();
+        var tocModels = models.OrderBy(f => f.File.Split('/').Count());
+        var tocCache = new Dictionary<string, TocItemViewModel>();
+        models.ForEach(model =>
+        {
+            tocCache.Add(model.Key.Replace("\\", "/").Replace("/" + tocFileName, string.Empty), (TocItemViewModel)model.Content);
+        });
+
+        // The list of models would contain all toc.yml including ones that are outside docfx base directory.
+        // Filter get the root toc
+        var rootTocModel = tocModels.Where(m =>
+            !m.LocalPathFromRoot.Contains("..")).OrderBy(f => f.LocalPathFromRoot.Split('/').Count()).FirstOrDefault();
+
+        // If there is no toc along side docfx.json, skip tocgen - validate behavior with yuefi
+        if (rootTocModel != null && rootTocModel.LocalPathFromRoot.Equals(tocFileName))
```

Comment:
`auto` should allow gen TOC based on the directory structure for any TOC file, not just the root

Reply:
i am glad i paused to ask this. what would be the scenario where if the root toc.yml is not present and the auto toc still need to populate the toc.yml in sub folders ?